src/misc/lv_mem.c

```diff
@@ -204,6 +204,39 @@ void * lv_mem_realloc(void * data_p, size_t new_size)
     return new_p;
 }
 
+bool lv_mem_realloc_safe(void ** data, size_t new_size)
+{
+    void * new_p = NULL;
+
+    MEM_TRACE("reallocating %p with %lu size", *data, (unsigned long)new_size);
+
+    if((*data == NULL) && (0 < new_size)) {
+        *data = lv_mem_alloc(new_size);
+        return false;
```

Comment:
What about returning `LV_RES_OK/INV`? It's cleaner then `true/false`.

Reply:
It's ok, I was thinking on returning the status of the allocation. `LV_RES_OK` if the allocation was done, `LV_RES_INV` otherwise. Are those enums already available?

Reply:
Resolved with 48c3333